### PR TITLE
feat: add checkbox component

### PR DESCRIPTION
**Pull request by ngochuyduong**
Custom checkbox component
uncheck state

check state

### CODE REVIEW

File: src/Checkbox/Checkbox.tsx [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+        {renderCheckIcon()}
+        {renderCheckboxText()}
+      </TouchableComponent>
+    )
+  },
+)
+
+export default Checkbox
+
+const styles = StyleSheet.create<any>({
```
Comment (TranQuangPhi) [inline src/Checkbox/Checkbox.tsx:199]: Please follow coding convention
https://trello.com/c/ClAU12Cu/43-coding-convention

File: src/Checkbox/Checkbox.tsx [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+
+    const renderCheckIcon = () => {
+      const checkStatus = disableBuiltInState ? isChecked! : checked
+      return (
+        <Animated.View
+          style={[
+            {transform: [{scale: bounceValue}]},
+            styles.iconContainer(size, checkStatus, fillColor, unfillColor),
+            iconStyle,
+            {opacity: disable ? disableOpacity : 1},
```
Comment (TranQuangPhi) [inline src/Checkbox/Checkbox.tsx:146]: not use style in-line

File: src/Checkbox/Checkbox.tsx [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+  unfillColor?: string
+  disable?: boolean
+  disableOpacity?: number
+  disableText?: boolean
+  bounceEffect?: number
+  bounceFriction?: number
+  useNativeDriver?: boolean
+  disableBuiltInState?: boolean
+  ImageComponent?: any
+  TouchableComponent?: any
```
Comment (TranQuangPhi) [inline src/Checkbox/Checkbox.tsx:37]: avoid using type "any"

File: src/Checkbox/Checkbox.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import React from 'react'
+import type {ComponentMeta, ComponentStory} from '@storybook/react'
+
+import Checkbox from './Checkbox'
+
+export default {
+  title: 'components/MyButton',
```
Comment (loc-nguyenthien) [inline src/Checkbox/Checkbox.stories.tsx:7]: Change title name

File: src/Checkbox/Checkbox.tsx [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+  textContainerStyle?: CustomStyleProp
+  checkIconImageSource?: ImageSourcePropType
+  onPress?: (checked: boolean) => void
+}
+
+interface ICheckboxMethods {
+  onHandlePress: () => void
+}
+
+const defaultCheckImage = require('./check.png')
```
Comment (loc-nguyenthien) [inline src/Checkbox/Checkbox.tsx:60]: create an `assets` folder for images

**Review by loc-nguyenthien — CHANGES_REQUESTED** [review]

File: src/Checkbox/Checkbox.tsx (hunk 2 above)
Comment (loc-nguyenthien) [reply src/Checkbox/Checkbox.tsx:149]: This is not consider in-line styles

File: src/Checkbox/Checkbox.stories.tsx (hunk 4 above)
Comment (loc-nguyenthien) [reply src/Checkbox/Checkbox.stories.tsx]: Done

File: src/Checkbox/Checkbox.tsx (hunk 5 above)
Comment (loc-nguyenthien) [reply src/Checkbox/Checkbox.tsx]: Done

**Review by loc-nguyenthien — APPROVED** [review]

**Review by TranQuangPhi — COMMENTED** [review]
please write test code for this component

File: src/Checkbox/Checkbox.tsx [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+      return (
+        (!disableText || checkDisableTextType) &&
+        (textComponent || (
+          <View
+            style={StyleSheet.flatten([
+              styles.textContainer,
+              textContainerStyle,
+              {opacity: disable ? disableOpacity : 1},
+            ])}>
+            <Text style={textStyle}>{text}</Text>
```
Comment (huydosgtech) [inline src/Checkbox/Checkbox.tsx:180]: should we check null at text?

Comment (ngochuyduong) [reply src/Checkbox/Checkbox.tsx:180]: no need to check

File: src/Checkbox/Checkbox.tsx [hunk 7]
```diff
@@ -0,0 +1,238 @@
...
+  StyleSheet,
+} from 'react-native'
+type CustomStyleProp = StyleProp<ViewStyle> | Array<StyleProp<ViewStyle>>
+type CustomTextStyleProp = StyleProp<TextStyle> | Array<StyleProp<TextStyle>>
+type CustomImageStyleProp = StyleProp<ImageStyle> | Array<StyleProp<ImageStyle>>
+type BaseTouchableProps = Pick<
+  TouchableWithoutFeedbackProps,
+  Exclude<keyof TouchableWithoutFeedbackProps, 'onPress'>
+>
+export interface ICheckboxProps extends BaseTouchableProps {
```
Comment (loc-nguyenthien) [inline src/Checkbox/Checkbox.tsx:23]: Please add some descriptions about these props

**Review by loc-nguyenthien — APPROVED** [review]

File: src/Checkbox/Checkbox.tsx (hunk 1 above)
Comment (ngochuyduong) [reply src/Checkbox/Checkbox.tsx]: updated

File: src/Checkbox/Checkbox.tsx (hunk 2 above)
Comment (ngochuyduong) [reply src/Checkbox/Checkbox.tsx:186]: not consider in-line styles

File: src/Checkbox/Checkbox.tsx (hunk 1 above)
Comment (TranQuangPhi) [reply src/Checkbox/Checkbox.tsx]: I mean use styled component

File: src/Checkbox/Checkbox.tsx (hunk 3 above)
Comment (ngochuyduong) [reply src/Checkbox/Checkbox.tsx]: updated

File: src/Checkbox/Checkbox.tsx (hunk 1 above)
Comment (ngochuyduong) [reply src/Checkbox/Checkbox.tsx]: Will update when styled component PR merge

File: src/Checkbox/Checkbox.tsx [hunk 8]
```diff
@@ -0,0 +1,274 @@
...
+            ])}>
+            <Text testID="text" style={textStyle}>
+              {text}
+            </Text>
+          </View>
+        ))
+      )
+    }
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (ThinhKimVo) [inline src/Checkbox/Checkbox.tsx:225]: why you need to ignore this eslint rule?

Comment (ngochuyduong) [reply src/Checkbox/Checkbox.tsx:225]: This branch don't have proper eslint config before, i will merge and remove it

File: src/Checkbox/Checkbox.tsx [hunk 9]
```diff
@@ -0,0 +1,274 @@
...
+          toValue: bounceEffectOut,
+          velocity: bounceVelocityOut,
+          bounciness: bouncinessOut,
+          useNativeDriver,
+        }),
+      ]).start()
+    }
+
+    const renderCheckIcon = () => {
+      const checkStatus = disableBuiltInState ? isChecked! : checked
```
Comment (anhle10051996) [inline src/Checkbox/Checkbox.tsx:171]: isChecked cannot be null or undefined ?

Comment (ngochuyduong) [reply src/Checkbox/Checkbox.tsx:171]: you mean

Comment (anhle10051996) [reply src/Checkbox/Checkbox.tsx:171]: write "isChecked!" you are telling to the compiler that isCheck variable cannot be null or undefined.

**Pull request closed without merging**